File: tests/integration_tests/tutorial_en/programs/tokens_en.py
# tokens 1 to 5

def collect_all():
    while object_here():
        take()
        move()

def put_all_down_and_move():
    while carries_object():
        put()
    move()
    
while not at_goal():
    if object_here():
        collect_all()
    elif carries_object():
        put_all_down_and_move()
    else:
        move()

File: tests/integration_tests/tutorial_en/programs/tokens_en.js
// tokens 1 to 5

function collect_all() { 
    while (object_here()) { 
        take(); 
        move(); 
    }
}

function put_all_down_and_move() { 
    while (carries_object()) { 
        put(); 
    }
    move(); 
}
    
while (!at_goal()) { 
    if (object_here()) { 
        collect_all(); 
    }
    else if (carries_object()) { 
        put_all_down_and_move(); 
    }
    else {
        move(); 
    }
}
